src/drivers/serac.cpp

```diff
@@ -15,113 +15,97 @@
 #include <fstream>
 #include <iostream>
 #include <memory>
+#include <string>
 
-#include "CLI11/CLI11.hpp"
+#include "axom/core.hpp"
 #include "coefficients/loading_functions.hpp"
 #include "coefficients/traction_coefficient.hpp"
+#include "infrastructure/cli.hpp"
 #include "infrastructure/initialize.hpp"
+#include "infrastructure/input.hpp"
 #include "infrastructure/logger.hpp"
 #include "infrastructure/terminator.hpp"
 #include "mfem.hpp"
 #include "numerics/mesh_utils.hpp"
 #include "physics/nonlinear_solid.hpp"
+#include "physics/utilities/equation_solver.hpp"
 #include "serac_config.hpp"
 
-int main(int argc, char* argv[])
-{
-  auto [num_procs, rank] = serac::initialize(argc, argv);
+namespace serac {
 
+//------- Input file -------
+
+void defineInputFileSchema(std::shared_ptr<axom::inlet::Inlet> inlet, int rank)
+{
   // mesh
-  std::string mesh_file = std::string(SERAC_REPO_DIR) + "/data/beam-hex.mesh";
+  inlet->addString("mesh", "Path to Mesh file")->required(true);
 
-  // serial and parallel refinement levels
-  int ser_ref_levels = 0;
-  int par_ref_levels = 0;
+  // Simulation time parameters
+  inlet->addDouble("t_final", "Final time for simulation.")->defaultValue(1.0);
+  inlet->addDouble("dt", "Time step.")->defaultValue(0.25);
 
-  // polynomial interpolation order
-  int order = 1;
+  // Refinement levels
+  inlet->addInt("ser_ref_levels", "Number of times to refine the mesh uniformly in serial.")->defaultValue(0);
+  inlet->addInt("par_ref_levels", "Number of times to refine the mesh uniformly in parallel.")->defaultValue(0);
 
-  // Solver parameters
-  serac::NonlinearSolverParameters nonlin_params;
-  nonlin_params.rel_tol     = 1.0e-2;
-  nonlin_params.abs_tol     = 1.0e-4;
-  nonlin_params.max_iter    = 500;
-  nonlin_params.print_level = 0;
+  // Polynomial interpolation order
+  inlet->addInt("order", "Order degree of the finite elements.")->defaultValue(1);
 
-  serac::LinearSolverParameters lin_params;
-  lin_params.rel_tol     = 1.0e-6;
-  lin_params.abs_tol     = 1.0e-8;
-  lin_params.max_iter    = 5000;
-  lin_params.print_level = 0;
+  // Physics
+  serac::NonlinearSolid::defineInputFileSchema(inlet);
 
-  // solver input args
-  bool gmres_solver = true;
-  bool slu_solver   = false;
+  // Verify input file
+  if (!inlet->verify()) {
+    SLIC_ERROR_ROOT(rank, "Input deck failed to verify.");
```

Comment:
Should we call them input files or input decks? Decks is the historical term, but it is quickly becoming obsolete.

Reply:
Does the term input "deck" actually from a collection of (punch) cards?

Reply:
yes its.. i try to catch myself whenever i call it a deck and move forward with the less lab specific name (input file).  I'll do another round and switch them.